modules.md

```diff
@@ -0,0 +1,233 @@
+# Module system extensions and assemblies
+
+This doc describes possible extensions to the current `import`/`export` syntax for modules. This includes support for assemblies.
+
+This doc is fairly detailed, but that is mostly to prove out the designs and ensure internal consistency, etc. Consider all syntax to be a strawman for iteration and the whole doc to be a starting point for discussing the design rather than a complete proposal.
+
+The doc is ordered by requirements, rather than a logical breakdown of the complete feature. Much is left unspecified for now, especially where I don't think it will affect other design choices. What is spec'ed and what isn't should be understood as a comment on overall prioritisation.
+
+Previous discussion:
+
+- [Initial modules proposal (`use` statements)](https://github.com/KittyCAD/modeling-app/issues/4080)
+- [Implementation of initial module system](https://github.com/KittyCAD/modeling-app/pull/4149)
+
+## Import all items from a module
+
+Importing all items (aka glob imports) is primarily a convenience so users don't need to make a long list of names in an `import` expression. I think this will get more important with some ideas I have for modularising std, but more on that elsewhere.
+
+The basic idea is we allow `import * from "foo.kcl"` which imports all `export`ed names from foo.kcl into the namespace of the current file.
+
+There are downsides to allowing this: primarily if an exported name is added to foo.kcl then that can cause an error in the importing file if the name already exists (which may be from another import, e.g., `import * from "bar.kcl"`). It's also a bit harder to follow the code without tooling (you can't tell at a glance where a name is defined).
+
+Syntax: `*` is common in many PLs, but is a bit opaque for new programmers. A keyword like `all`, `everything`, or `any` would be better, but that is one less name we can all use for programming (we can't use a contextual keyword here). I think it would be a bit too subtle to use no names to mean all names (e.g., `import from "foo.kcl"`).
```

Comment:
What about `import all symbols from "foo.kcl"`? It would be unambiguous because if you're trying to import two parts called "all" and "symbols" you would need a comma between them.